### PR TITLE
Fix(analysis): Set limit for reports export

### PR DESCRIPTION
**Pull request by aziiee**

### CODE REVIEW

File: app/analysis/index/controller.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ const serializeMoment = momentObject =>
 const deserializeMoment = momentString =>
   (momentString && moment(momentString, DATE_FORMAT)) || null;
 
+const EXPORT_LIMIT = 50000;
```
Comment (aziiee) [inline app/analysis/index/controller.js:36]: @sbor23 Please confirm the export limit.

**Comment by sbor23** [comment]
please wait before you merge this. At the moment 100'000 would work but I'm looking to into solving this problem completely using annotation. So in the best case we don't need this at all.

Anyway this number shouldn't be hardcoded but made configurable, also in the frontend if possible.
System performance differs a lot and there would be no guarantee that something that works locally would also work the test/prod server.

**Comment by sbor23** [comment]
See https://github.com/adfinis-sygroup/timed-backend/pull/541 for the backend part which is configurable.

File: app/analysis/index/template.hbs [hunk 2]
```diff
@@ -305,6 +305,8 @@
             type="button"
             data-download-count="{{download.performCount}}"
             class="btn btn-primary {{if (and (eq download.last.url link.url) (eq download.last.params link.params) download.isRunning) 'loading'}}"
+            disabled={{exportLimitExceeded}}
+            title={{if exportLimitExceeded "Please use filters to reduce the amount of reports to export."}}
```
Comment (anehx) [inline app/analysis/index/template.hbs:309]: Include the export maximum here so the user knows it

Comment (aziiee) [reply app/analysis/index/template.hbs]: @anehx done

File: app/analysis/index/controller.js [hunk 3]
```diff
@@ -184,8 +184,20 @@ const AnalysisController = Controller.extend(AnalysisQueryParams.Mixin, {
...
   ),
 
+  exportLimitExceeded: computed("totalItems", function() {
+    return this.get("totalItems") > this.get("exportLimit");
+  }),
+
+  exportLimitMessage: computed("exportLimit", function() {
+    return `The export limit is ${this.get(
+      "exportLimit"
+    )}. Please use filters to reduce the amount of reports.`;
```
Comment (anehx) [inline app/analysis/index/controller.js:194]:
```suggestion
    return `The export limit is ${this.exportLimit}. Please use filters to reduce the amount of reports.`;
```

`.get` is not necessary nowadays

Comment (aziiee) [reply app/analysis/index/controller.js]: should be okay now

**Pull request merged**